python/ray/serve/_private/replica.py

```diff
@@ -311,8 +314,12 @@ async def is_initialized(
             # Unused `_after` argument is for scheduling: passing an ObjectRef
             # allows delaying reconfiguration until after this call has returned.
             try:
-                await self._initialize_replica()
-                metadata = await self.reconfigure(deployment_config)
+                async with self._replica_init_lock:
+                    if not self._replica_init_finish:
+                        await self._initialize_replica()
+                if deployment_config:
+                    await self.reconfigure(deployment_config)
```

Comment:
do we need to worry about concurrency issues with running `reconfigure`?

Reply:
Since this call to reconfigure is something that is meant to be called once directly following replica initialization, would it be safer to just place it inside the lock?

Reply:
good point

Reply:
Moved under the lock cloure.